### PR TITLE
[MAYA] Configure pyblish behaviour with environment variable

### PR DESCRIPTION
**Pull request by jlorrain**
The goal is to provide environment variables to preset pyblish behaviour before launching it.

Those variables can be used with the maya "Script Menu Definition" for example to create shortcut for the operator to quickly publish, review, or render his work without having to activate/deactivate instances or plugins within pyblish

### Work to come
- [x] "PYBLISH_FAMILY_WHITELIST", force pyblish to only considere whitelisted families
- [x] "PYBLISH_DEFAULT_INTENT", use this value as default intent
- [x] "PYBLISH_TITLE", Override pyblish window title

### CODE REVIEW

**Comment by iLLiCiTiT** [comment]
I don't have problem to have these features but I don't think it should be defined by environment variables.

**Comment by mkolar** [comment]
>context inside host may change and you may "not want" it.

I think the intent here is that value of that variable would change before invoking publisher based on what "menu option" user chooses. So you could have, not just `publish` but for example, `publish render only` and when you run that only renders are enabled. It is pretty much saving a click or two.

Having it in the setting instead of env var, might actually make it less flexible in this case.

**Comment by iLLiCiTiT** [comment]
> I think the intent here is that value of that variable would change before invoking publisher based on what "menu option" user chooses.

I understand but that's the issue I see why not use environment variables. They're easily "accidentally set", not "unset" or passed from/to different process. I would preffer some global variables that are at least process specific and can't be passed to different processes or farm or anywhere. Those global variables should be known for standard publish action as should probably unset values that were set with different action when is triggered (to cancel behavior of custom action).

File: openpype/tools/pyblish_pype/app.py [hunk 1]
```diff
@@ -98,7 +98,11 @@ def show(parent=None):
 
         self._window.show()
         self._window.activateWindow()
-        self._window.setWindowTitle(settings.WindowTitle)
+        env_title = os.getenv("PYBLISH_TITLE")
+        if env_title:
+            self._window.setWindowTitle(env_title)
+        else:
+            self._window.setWindowTitle(settings.WindowTitle)
```
Comment (BigRoy) [inline openpype/tools/pyblish_pype/app.py:105]: Why need the environment variable if you could also tweak `settings.WindowTitle`? Or are you running pyblish_pype in a subprocess?

**Comment by mkolar** [comment]
@jlorrain do you think this is reviewable? Trying to clean up old PRs and this one does have some value, albeit I'm not 100% convinced. 

@antirotor @BigRoy any opinions on this in general?

**Comment by BigRoy** [comment]
Hard to say - I feel it's a workaround for something that might be better solved differently.

**Pull request closed without merging**